detection_rules/etc/version.lock.json

```diff
@@ -153,7 +153,7 @@
   "035889c4-2686-4583-a7df-67f89c292f2c": {
     "min_stack_version": "8.3",
     "previous": {
-      "7.16": {
+      "8.3": {
```

Comment:
Something seems strange about this. The min_stack_version is 8.3 and it has a previous entry of the same version?

Reply:
The initial CLI command was developed and merged 06/02/2022 which was tested then, but the schema for version lock file allowed a full semantic version. This then changed 06/07/2022 where the schema only supported major and minor semantic versions, but the CLI command was not tested and thus up to this point needed to be adjusted.

[CLI command added](https://github.com/elastic/detection-rules/pull/2013/files#diff-8dcef89fe672ab07924dae360eeb04a04b69fb7d66c60f702359920769399827)
[Schema updated](https://github.com/elastic/detection-rules/pull/2021/files#diff-1b7f9f4101b33ab4640395a3ac7fe8c8ba7d9a3d6f9a4844f9b19458792897a5)

Reply:
https://github.com/elastic/detection-rules/blob/main/detection_rules/devtools.py#L910-L911

Reply:
good catch

https://github.com/elastic/detection-rules/blob/7725e3212645cfdc4d03f30c7402dae0703176a4/detection_rules/devtools.py#L920

looks like this needs to be updated

Reply:
if latest == current, drop

Reply:
The latest previous (forked) should never match the current for the version lock file. In our case, we need to check every previous (forked) version is greater than or equal to the supplied min_version we want to trim for. If so, that previous version should be added back to the lock version.